.github/workflows/android.yml

```diff
@@ -18,21 +21,41 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
+    - name: Enable KVM
+      run: |
+        echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+        sudo udevadm control --reload-rules
+        sudo udevadm trigger --name-match=kvm
+
     - uses: actions/cache@v4
       id: openssl
       with:
         path: openssl
-        key: ${{ runner.os }}-android-openssl-${{ env.openssl }}
+        key: ${{ runner.os }}-android-${{ env.abi }}-openssl-${{ env.openssl }}
 
     - name: "build openssl"
       if: steps.openssl.outputs.cache-hit != 'true'
       run: |
         wget -q https://www.openssl.org/source/openssl-$openssl.tar.gz
         tar -xzf openssl-$openssl.tar.gz
         mv openssl-$openssl openssl
-        cd openssl && ANDROID_NDK_ROOT=$ANDROID_NDK_LATEST_HOME PATH=$ANDROID_NDK_LATEST_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH ./Configure android-arm64 no-shared no-tests -U__ANDROID_API__ -D__ANDROID_API__=21 && PATH=$ANDROID_NDK_LATEST_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH make build_libs && cd ..
+        cd openssl && ANDROID_NDK_ROOT=$ANDROID_NDK_LATEST_HOME PATH=$ANDROID_NDK_LATEST_HOME/$toolchain/bin:$PATH ./Configure android-$abi no-shared no-tests -U__ANDROID_API__ -D__ANDROID_API__=$api && PATH=$ANDROID_NDK_LATEST_HOME/$toolchain/bin:$PATH make build_libs && cd ..
 
     - name: build
       run: |
-        cmake -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK_LATEST_HOME/build/cmake/android.toolchain.cmake -DANDROID_ABI="arm64-v8a" -DANDROID_PLATFORM=android-21 -DOPENSSL_ROOT_DIR=openssl -DOPENSSL_INCLUDE_DIR=openssl/include -DOPENSSL_CRYPTO_LIBRARY=openssl/libcrypto.a -DOPENSSL_SSL_LIBRARY=openssl/libssl.a .
+        cmake -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK_LATEST_HOME/build/cmake/android.toolchain.cmake -DANDROID_ABI="$abi" -DANDROID_PLATFORM=android-$api -DOPENSSL_ROOT_DIR=openssl -DOPENSSL_INCLUDE_DIR=openssl/include -DOPENSSL_CRYPTO_LIBRARY=openssl/libcrypto.a -DOPENSSL_SSL_LIBRARY=openssl/libssl.a -DUSE_UNIXSOCK=OFF .
```

Comment:
Android does not support pathname addressed unix sockets, only abstract addressed. See https://www.man7.org/linux/man-pages/man7/unix.7.html for the distinction.

I will disable it from Android for now since the existing API has no way to specify an abstract address due to the `\0` start byte. We could consider a future enhancement to add support for `abstract:` socket addresses prefixes, similarly to [`socat`](https://man.archlinux.org/man/socat1.1.en#ABSTRACT)..